vendor/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -257,6 +257,8 @@ type Config struct {
 
 	// StorageVersionManager holds the storage versions of the API resources installed by this server.
 	StorageVersionManager storageversion.Manager
+
+	EnableWatermarkMaintenance bool
```

Comment:
You probably want this to be `DisableWatermarkMaintenance` so the default is enabled which is what's currently expected

Reply:
I think that the default should be the current behavior (enabled), not disabled. I think that changes like this should almost always be initially opt-in (especially for changes going into upstream like this).